presto-main/src/main/java/com/facebook/presto/sql/relational/SqlToRowExpressionTranslator.java

```diff
@@ -160,13 +161,16 @@ public static RowExpression translate(
         private final Map<NodeRef<Expression>, Type> types;
         private final TypeManager typeManager;
         private final TimeZoneKey timeZoneKey;
+        @Deprecated
```

Comment:
I do not feel as this should be deprecated. It is a timestamp which is deprecated for users to use, so they should use a new semantic. Here you do not have an alternative and using this switch is completely valid thing here. What do you think? I have seen other places like that, if you agree with me I can comment there as well.